qiskit_research/utils/pauli_twirling.py

```diff
@@ -43,31 +70,94 @@
     "secr": Pauli("XZ"),
 }
 
+
+def match_global_phase(a, b):
+    """Phase the given arrays so that their phases match at one entry.
+
+    Args:
+        a: A Numpy array.
+        b: Another Numpy array.
+
+    Returns:
+        A pair of arrays (a', b') that are equal if and only if a == b * exp(i phi)
+        for some real number phi.
+    """
+    if a.shape != b.shape:
+        return a, b
+    # use the largest entry of one of the matrices to maximize precision
+    index = max(np.ndindex(*a.shape), key=lambda i: abs(b[i]))
+    phase_a = cmath.phase(a[index])
+    phase_b = cmath.phase(b[index])
+    return a * cmath.rect(1, -phase_a), b * cmath.rect(1, -phase_b)
+
+
+def allclose_up_to_global_phase(op1, op2, tol=1e-15):
```

Comment:
Nit: Please match the API of [`numpy.allclose`](https://numpy.org/devdocs/reference/generated/numpy.allclose.html), and have the inputs be plain Numpy arrays instead Qiskit Operators.

```python
def allclose_up_to_global_phase(a, b, rtol=1e-05, atol=1e-08, equal_nan=False):
```